src/polyswarm_api/types/result.py

```diff
@@ -146,10 +146,12 @@ def __init__(self, rules, result, polyswarm=None):
         super(HuntSubmissionResult, self).__init__(result, polyswarm)
         self.rules = rules
 
-        if self.status_code // 100 != 2:
+        if self.status_code == 400:
+            self.result = 'Syntax error in submission. Please check your rules, or install the yara-python package for more details.'
+        elif self.status_code // 100 != 2:
             raise self._bad_status_exception
```

Comment:
A new exception should be generated here instead of using an existing one. This will make the new exception to preserve the stack trace of this point in the code. There is no reason to have a pre-created exception. We can, however, have a pre-created exception message if necessary. Still, the exception should preferably be instantiated here.

Reply:
This is not a "pre-created exception". It's a `@property` that generates an exception that's used a bunch of places.

Reply:
See https://github.com/polyswarm/polyswarm-api/blob/572ac6d4f2534a1f620ec90a2c785257eab2f791/src/polyswarm_api/types/result.py#L51

Reply:
I would prefer that we have a `def _bad_status_message(self): ` and raise the exception on the spot. It is not clear really what is happening from looking at the code. I think 
```python
raise exceptions.ServerErrorException(self._bad_status_message)
```
would make it much more clear (it also would not include the property call in the stack).

Reply:
Agree: https://github.com/polyswarm/polyswarm-api/pull/71